specifyweb/specify/views.py

```diff
@@ -509,6 +509,32 @@ def fix_orderings(base_model: Table, new_record_data):
                     resources.insert(0, record)
             new_record_data[field_name] = resources
 
+def fix_record_data(new_record_data, model_name, old_model_ids, new_model_id):
+    final = {}
+    for key, value in new_record_data.items():
+
+        if isinstance(value, str) and model_name.lower() in value:
```

Comment:
wouldn't this break if value is a random text field and someone put in `agent` text?